components/common/IdentityForm.vue

```diff
@@ -37,82 +25,192 @@
+      :identity="identity"
+      :socials="socialTabs"
+      :is-mobile="isMobile"
+      @confirm="submit"
+      @close="isConfirmModalActive = false" />
+
+    <TransactionLoader
+      v-model="isLoaderModalVisible"
+      :status="status"
+      :action-title="$t('identity.create')"
+      :transaction-id="transactionValue"
+      :is-mobile="isMobile"
+      @close="isLoaderModalVisible = false" />
   </section>
 </template>
 
 <script lang="ts" setup>
 import { notificationTypes, showNotification } from '@/utils/notification'
-import { NeoField, NeoIcon, NeoInput, NeoTooltip } from '@kodadot1/brick'
+import {
+  NeoButton,
+  NeoField,
+  NeoIcon,
+  NeoInput,
+  NeoTooltip,
+} from '@kodadot1/brick'
 import type { IdentityFields } from '@/composables/useIdentity'
-const Auth = defineAsyncComponent(() => import('@/components/shared/Auth.vue'))
-const BasicInput = defineAsyncComponent(
-  () => import('@/components/shared/form/BasicInput.vue')
-)
-const Loader = defineAsyncComponent(
-  () => import('@/components/shared/Loader.vue')
-)
-const Money = defineAsyncComponent(
-  () => import('@/components/shared/format/Money.vue')
-)
-const SubmitButton = defineAsyncComponent(
-  () => import('@/components/base/SubmitButton.vue')
-)
-
-const { $i18n } = useNuxtApp()
+import PillTabs, { PillTab } from '@/components/shared/PillTabs.vue'
+import IdentityConfirmModal from '@/components/common/identity/IdentityConfirmModal.vue'
+import TransactionLoader from '@/components/shared/TransactionLoader.vue'
 import { useIdentityStore } from '@/stores/identity'
+import BasicInput from '@/components/shared/form/BasicInput.vue'
+import Money from '@/components/shared/format/Money.vue'
+import { useFiatStore } from '@/stores/fiat'
+import { calculateUsdFromToken } from '@/utils/calculation'
+import format from '@/utils/format/balance'
 
+const { $i18n } = useNuxtApp()
 const { apiInstance } = useApi()
-const { accountId, balance } = useAuth()
+const { accountId } = useAuth()
+const { decimals, unit, name: chainName } = useChain()
 const { urlPrefix } = usePrefix()
+const { fetchFiatPrice, getCurrentTokenValue } = useFiatStore()
 const identityStore = useIdentityStore()
+const { identity: identityData } = useIdentity({
+  address: accountId,
+})
 const { howAboutToExecute, isLoading, initTransactionLoader, status } =
   useMetaTransaction()
+
 const identity = ref<IdentityFields>({})
 const deposit = ref('0')
 const inputLengthLimit = ref(32)
 
-const { identity: identityData } = useIdentity({
-  address: accountId,
+const isConfirmModalActive = ref(false)
+const isLoaderModalVisible = ref(false)
+const transactionValue = ref('')
+
+enum IdentitySocialField {
+  Riot = 'riot',
+  Twitter = 'twitter',
+}
+
+const socialTabs = ref<PillTab[]>([
+  {
+    label: 'Riot',
+    value: IdentitySocialField.Riot,
+  },
+  {
+    label: 'Twitter',
+    icon: { name: 'fa-x-twitter', pack: 'fa-brands' },
+    value: IdentitySocialField.Twitter,
+  },
+])
```

Comment:
it looks to me that if you used map instead of array the handling of it inside `IdentityConfirmModal` would be quite simplified

something like:
```js
{
Riot: 
    {
      label: 'Riot',
      value: IdentitySocialField.Riot,
    }
}
```